docs/source/55_api.md

```diff
@@ -257,27 +257,30 @@ bypassing the ADAM APIs.
 
 Another useful API implemented in ADAM is the RegionJoin API, which joins two
 genomic datasets that contain overlapping regions. This primitive is useful for
-a number of applications including variant calling (identifying all of the reads 
-that overlap a candidate variant), coverage analysis (determining the coverage 
-depth for each region in a reference), and INDEL realignment (identify INDELs 
+a number of applications including variant calling (identifying all of the reads
+that overlap a candidate variant), coverage analysis (determining the coverage
+depth for each region in a reference), and INDEL realignment (identify INDELs
 aligned against a reference).
 
-There are two overlap join implementations available in ADAM: 
-BroadcastRegionJoin and ShuffleRegionJoin. The result of a ShuffleRegionJoin 
-is identical to the BroadcastRegionJoin, however they serve different 
+There are two overlap join implementations available in ADAM:
+BroadcastRegionJoin and ShuffleRegionJoin. The result of a ShuffleRegionJoin
+is identical to the BroadcastRegionJoin, however they serve different
 purposes depending on the content of the two datasets.
 
-The ShuffleRegionJoin is at its core a distributed sort-merge overlap join. 
-To ensure that the data is appropriately colocated, we perform a copartition 
-on the right dataset before the each node conducts the join locally. The 
-BroadcastRegionJoin performs an overlap join by broadcasting a copy of the 
-entire left dataset to each node. ShuffleRegionJoin should be used if the right 
-dataset is too large to send to all nodes and both datasets have low 
-cardinality. The BroadcastRegionJoin should be used when you are joining a 
-smaller dataset to a larger one and/or the datasets in the join have high 
-cardinality.
-
-Another important distinction between ShuffleRegionJoin and 
+The ShuffleRegionJoin is at its core a distributed sort-merge overlap join.
+To ensure that the data is appropriately colocated, we perform a copartition
+on the right dataset before the each node conducts the join locally.
+ShuffleRegionJoin should be used if the right dataset is too large to send to
+all nodes and both datasets have high cardinality. Because of the flexibility
+of the sort-merge paradigm, some joins can only be performed as a
+ShuffleRegionJoin (e.g. full outer join).
+
+The BroadcastRegionJoin performs an overlap join by broadcasting a copy of the
+entire left dataset to each node. The BroadcastRegionJoin should be used when
+you are joining a smaller dataset to a larger one and/or the datasets in the
```

Comment:
More so, I'd say the broadcast region join should be used when you have a dataset that is small enough to be collected and broadcast out, the larger side of the join is unsorted and either the data is so skewed that it is hard to load balance, the data is too large to be worth shuffling, or you don't want sorted output.